augur/tracers/two_point.py

```diff
@@ -221,3 +221,34 @@ def to_dict(self):
         dict_out = self.__dict__
         dict_out['kind'] = 'WLSource'
         return dict_out
+
+
+class ZDistFromFile(ZDist):
+    """
+    Class to pass an arbitrary redshift distribution to a tracer
+    """
+    def __init__(self, input_file, ibin=None, format='npy', **kwargs):
+        """
+        Parameters:
+        -----------
+        input_file : str
+            Path to input file with z and dndz
+        """
+        if 'npy' in format:
+            data = np.load(input_file, allow_pickle=True)
+            self.z = data[()]['redshift_range']
+            if ibin is None:
+                raise ValueError('Expected bin number for this format')
+            else:
+                self.Nz = data[()]['bins'][ibin]
+        elif 'ascii' in format:
+            data = np.loadtxt(input_file)
+            self.z = data[:, 0]
+            if ibin is None:
+                self.Nz = data[:, 1]
+            else:
+                self.Nz = data[:, ibin+1]
+        else:
+            raise NotImplementedError('Only ascii files or npy files are currently supported')
```

Comment:
Shall we add a check to see if the number of dndz columns in the file matches the `nbins` keyword from the configuration files?

Reply:
I realised this is done by construction in the `generate.py` structure!